settings/OpeningPeriodForm/BigCalendarWrapper.js

```diff
@@ -122,7 +128,32 @@ class BigCalendarWrapper extends React.PureComponent {
       this.onCalendarChange(nextEvents);
     };
 
+  checkEventExistOrOverlap = (startTime, endTime) => {
+    const { events } = this.state;
+    return events.every(event => {
+      const { start, end } = event;
+      const startOfExistedEvent = moment(start);
+      const endOfExistedEvent = moment(end);
+      const startTimeMoment = moment(startTime);
+      const endTimeMoment = moment(endTime);
+      const duplicateEvent = startTimeMoment.isSame(startOfExistedEvent) && endTimeMoment.isSame(endOfExistedEvent);
```

Comment:
I see that things can be simplified. I am not sure it covers all cases there, so please verify whether it fits your needs.

```suggestion
import { range } from 'moment-range';

const existedEventRange = range([moment(start), moment(end)]);
const newEventRange = range([moment(startTime), moment(endTime)]);

return existedEventRange.overlaps(newEventRange);
```